File: backend/src/main/java/me/performancereservation/domain/file/FileRepository.java
package me.performancereservation.domain.file;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface FileRepository extends JpaRepository<File, Long> {

    @Query("SELECT f FROM File f WHERE f.id IN :fileIds")
    List<File> findAllById(@Param("fileIds") List<Long> fileIds);
}

File: backend/src/main/kotlin/me/performancereservation/domain/file/FileRepository.kt
package me.performancereservation.domain.file

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface FileRepository : JpaRepository<File, Long> {
    @Query("""
        SELECT f 
        FROM File f 
        WHERE f.id IN :fileIds
    """)
    fun findAllById(@Param("fileIds") fileIds: List<Long>): List<File>
}
